express/scripts/utils/pricing.js

```diff
@@ -0,0 +1,400 @@
+        `<strong>${plan.prefix}${plan.rawPrice[0]}</strong>`,
+      );
+
+      if (offer.basePriceCurrencyFormatted) {
+        plan.rawBasePrice = offer.basePriceCurrencyFormatted.match(/[\d\s,.+]+/g);
+        plan.formattedBP = offer.basePriceCurrencyFormatted.replace(
+          plan.rawBasePrice[0],
+          `<strong>${plan.prefix}${plan.rawBasePrice[0]}</strong>`,
+        );
+      }
+      plan.y2p = offer.y2p;
+    }
+    window.pricingPlans[planUrl] = plan;
+  }
+  return plan;
+}
+
+function replaceUrlParam(url, paramName, paramValue) {
+  const params = url.searchParams;
+  params.set(paramName, paramValue);
+  url.search = params.toString();
+  return url;
+}
+
+// TODO probably want to replace / merge this with new getEnv method
+export function getHelixEnv() {
+  let envName = sessionStorage.getItem('helix-env');
+  if (!envName) {
+    envName = 'stage';
+    if (window.spark?.hostname === 'www.adobe.com') envName = 'prod';
+  }
+  const envs = {
+    stage: {
+      commerce: 'commerce-stg.adobe.com',
+      adminconsole: 'stage.adminconsole.adobe.com',
+      spark: 'stage.projectx.corp.adobe.com',
+    },
+    prod: {
+      commerce: 'commerce.adobe.com',
+      spark: 'express.adobe.com',
+      adminconsole: 'adminconsole.adobe.com',
+    },
+  };
+  const env = envs[envName];
+
+  const overrideItem = sessionStorage.getItem('helix-env-overrides');
+  if (overrideItem) {
+    const overrides = JSON.parse(overrideItem);
+    const keys = Object.keys(overrides);
+    env.overrides = keys;
+
+    for (const a of keys) {
+      env[a] = overrides[a];
+    }
+  }
+
+  if (env) {
+    env.name = envName;
+  }
+  return env;
+}
+
+export function buildUrl(optionUrl, country, language, offerId = '') {
+  const currentUrl = new URL(window.location.href);
+  let planUrl = new URL(optionUrl);
+
+  if (!planUrl.hostname.includes('commerce')) {
+    return planUrl.href;
+  }
+  planUrl = replaceUrlParam(planUrl, 'co', country);
+  planUrl = replaceUrlParam(planUrl, 'lang', language);
+  if (offerId) {
+    planUrl.searchParams.set(decodeURIComponent('items%5B0%5D%5Bid%5D'), offerId);
+  }
+  let rUrl = planUrl.searchParams.get('rUrl');
+  if (currentUrl.searchParams.has('host')) {
+    const hostParam = currentUrl.searchParams.get('host');
+    const { host } = new URL(hostParam);
+    if (host === 'express.adobe.com') {
+      planUrl.hostname = 'commerce.adobe.com';
+      if (rUrl) rUrl = rUrl.replace('express.adobe.com', hostParam);
+    } else if (host === 'qa.adobeprojectm.com') {
+      planUrl.hostname = 'commerce.adobe.com';
+      if (rUrl) rUrl = rUrl.replace('express.adobe.com', hostParam);
+    } else if (host.endsWith('.adobeprojectm.com')) {
+      planUrl.hostname = 'commerce-stg.adobe.com';
+      if (rUrl) rUrl = rUrl.replace('adminconsole.adobe.com', 'stage.adminconsole.adobe.com');
+      if (rUrl) rUrl = rUrl.replace('express.adobe.com', hostParam);
+    }
+  }
+
+  const env = getHelixEnv();
```

Comment:
@vhargrave do you think that it's safe to use `const { env } = getConfig();` here and then change the lines below to follow the `env` api? Or do we still need a decision function that looks at local storage to determine whether to use `getConfig().prod` or `getConfig().stage`...

Reply:
@jrrings I'm not exactly sure what you mean with the local storage, but I'd just rewrite this whole snippet to this. (Haven't tested it)
```
const env = getConfig().env.name;
  if (env && getConfig()[env].commerce && planUrl.hostname.includes('commerce')) planUrl.hostname = getConfig()[env].commerce;
  if (env && getConfig()[env].express && rUrl) {
    const url = new URL(rUrl);
    url.hostname = getConfig()[env].express;
    rUrl = url.toString();
  }
```

Reply:
ok cool! That's what I was thinking. In the current `getHelixEnv` function there is a check in localStorage, but I think this will be just fine.